src/lambdaisland/deep_diff2/strip.cljc

```diff
@@ -0,0 +1,43 @@
+(ns lambdaisland.deep-diff2.strip
+  "Provide API for manipulate the diff structure data "
+  (:require [clojure.walk :refer [postwalk]]
+            #?(:clj [lambdaisland.deep-diff2.diff-impl]
+               :cljs [lambdaisland.deep-diff2.diff-impl :refer [Mismatch Deletion Insertion]]))
+  #?(:clj (:import [lambdaisland.deep_diff2.diff_impl Mismatch Deletion Insertion])))
+
+(defn diff-item?
+  "Checks if x is a Mismatch, Deletion, or Insertion"
+  [x]
+  (or (instance? Mismatch x)
+      (instance? Deletion x)
+      (instance? Insertion x)))
+
+(defn extend-flatten
+  "Flatten, which can apply on hashmap"
+  [x]
+  (filter (complement coll?)
+          (rest (tree-seq coll? seq x))))
+
+(defn has-diff-item?
+  "Checks if there are any diff items in x or sub-tree of x"
```

Comment:
Maybe use `coll` instead of `x`?

Reply:
The `has-diff-item?` applies to the val item of a `map-entry`. It could be any type, so I choose `x`.